core/src/types/token.rs

```diff
@@ -401,6 +404,81 @@ impl DenominatedAmount {
             })
             .ok_or(AmountParseError::PrecisionOverflow)
     }
+
+    /// Convert this denominated amount into a plain amount by increasing its
+    /// precision to the given token's denomination and then taking the
+    /// significand.
+    pub fn to_amount(
+        self,
+        token: &Address,
+        storage: &impl StorageRead,
+    ) -> storage_api::Result<Amount> {
+        let denom = read_denom(storage, token)?.ok_or_else(|| {
+            storage_api::Error::SimpleMessage(
+                "No denomination found in storage for the given token",
+            )
+        })?;
+        self.increase_precision(denom)
+            .map_err(storage_api::Error::new)
+            .map(|x| x.amount)
+    }
+
+    /// Multiply this number by 10^denom and return the computed integer if
+    /// possible. Otherwise error out.
+    pub fn scale(
+        self,
+        denom: impl Into<u8>,
+    ) -> Result<Amount, AmountParseError> {
+        self.increase_precision(Denomination(denom.into()))
+            .map(|x| x.amount)
+    }
+
+    /// Checked multiplication. Returns `None` on overflow.
+    pub fn checked_mul(&self, rhs: DenominatedAmount) -> Option<Self> {
+        let amount = self.amount.checked_mul(rhs.amount)?;
+        let denom = self.denom.0.checked_add(rhs.denom.0)?.into();
+        Some(Self { amount, denom })
+    }
+
+    /// Checked subtraction. Returns `None` on overflow.
+    pub fn checked_sub(&self, mut rhs: DenominatedAmount) -> Option<Self> {
+        let mut lhs = *self;
+        if lhs.denom < rhs.denom {
+            lhs = lhs.increase_precision(rhs.denom).ok()?;
+        } else {
+            rhs = rhs.increase_precision(lhs.denom).ok()?;
+        }
+        let amount = lhs.amount.checked_sub(rhs.amount)?;
+        Some(Self {
+            amount,
+            denom: lhs.denom,
+        })
+    }
+
+    /// Checked addition. Returns `None` on overflow.
+    pub fn checked_add(&self, mut rhs: DenominatedAmount) -> Option<Self> {
+        let mut lhs = *self;
+        if lhs.denom < rhs.denom {
+            lhs = lhs.increase_precision(rhs.denom).ok()?;
+        } else {
+            rhs = rhs.increase_precision(lhs.denom).ok()?;
+        }
+        let amount = lhs.amount.checked_add(rhs.amount)?;
+        Some(Self {
+            amount,
+            denom: lhs.denom,
+        })
+    }
+
+    /// Returns the significand of this number
+    pub fn amount(&self) -> Amount {
+        self.amount
+    }
```

Comment:
this method might be a bit confusing, since it doesn't specify that the returned amount is not multiplied by the denomination. maybe mention this in the docstr.

also, I think this can become a `const fn`